metrics/ci_integrations/test/source/github_actions/config/validation_delegate/github_actions_source_validation_delegate_test.dart

```diff
@@ -430,5 +532,346 @@ void main() {
         expect(interactionResult.isSuccess, isTrue);
       },
     );
+
+    test(
+      ".validateJobName() returns a successful interaction with null result if the workflow runs with conclusion fetching failed",
+      () async {
+        whenFetchWorkflowRunsWithConclusion(workflowId).thenErrorWith();
+
+        final interactionResult = await delegate.validateJobName(
+          workflowId: workflowId,
+          jobName: jobName,
+        );
+
+        expect(interactionResult.isSuccess, isTrue);
+        expect(interactionResult.result, isNull);
```

Comment:
I think it would be better to split such tests into two separate tests testing the success of the interaction and the result value.